### PR TITLE
Do not update sysconfig when no IP is given in case of static bootproto

### PR DESCRIPTION
**Pull request by mchf**
bnc#1046735

### CODE REVIEW

File: src/include/network/lan/address.rb [hunk 1]
```diff
@@ -1482,33 +1482,32 @@ def AddressDialog
...
-                @hostname_initial,
-                Ops.get_string(@settings, "HOSTNAME", ""),
-                Ops.get_string(@settings, "IPADDR", "")
-              )
-            end
+          if (@hostname_initial != hostname && !hostname.empty?) || ip_changed
+            Host.Update(@hostname_initial, hostname, LanItems.ipaddr)
           end
-        else
+        elsif bootproto != "static"
```
Comment (jreidinger) [inline src/include/network/lan/address.rb]: so what happen if not ip is set?

Comment (mchf) [inline src/include/network/lan/address.rb]: hopefully nothing. Intention is not to update config bcs ```BOOTPROTO=static``` makes no sense without IP defined

Comment (jreidinger) [inline src/include/network/lan/address.rb]: so why not use return instead like
`return nil if bootproto == "static" && ipaddr.empty?`
and maybe log it out?

Comment (jreidinger) [inline src/include/network/lan/address.rb]: or to be correct `return ret`

Comment (jreidinger) [inline src/include/network/lan/address.rb]: and all in all. Is it valid at all? Maybe it is bug in validation of ip address widget, not?

**Comment by mchf** [comment]
Thanks for the review

**Pull request merged**